### ISSUE TITLE
data.all in local environment starts in a broken db state

### ISSUE THREAD

**Issue by zsaltys** [data-dot-all/dataall#788]
### Describe the bug

With the current 2.0 branch of data.all if we start data.all in a local dev environment then it will start with HALF the tables missing. This is the first issue and requires running alembic migrations to fix.

Upon running alembic migrations we find that they don't run with the error:

> sqlalchemy.exc.ProgrammingError: (pg.ProgrammingError) ERROR:  column "privateSubnetIds" of relation "vpc" already exists

coming from:

> INFO  [alembic.runtime.migration] Running upgrade bd271a2780b2 -> 5d5102986ce5, Add subnet ids columns

We can then attempt to drop all the tables with:

python backend/migrations/drop_tables.py

And run the migrations again but then data.all won't start because permission table will be missing permissions such as MANAGE_GLOSSARY, MANAGE_DATASETS

### How to Reproduce

Start data.all in dev environment from scratch and attempt to upgrade the local database with latest migrations

### Expected behavior

1. 2.0 branch local dev environment should start with all the tables installed
2. If I drop all the tables and attempt to rerun all the migrations, this should work and data.all should start without issues
3. https://awslabs.github.io/aws-dataall/deploy-locally/ should be updated to include explaining the alembic migrations and how to run them to update to the latest state and how to add new ones.
4. Ideally there should be tests that test that data.all can start with migrations rebuilt from scratch

### OS

NA

### Python version

NA

### AWS data.all version

2.0

**Comment by dlpzx** [comment]
Hi @zsaltys, thanks for opening the issue. I am trying to reproduce the problem.

**Comment by dlpzx** [comment]
## Test 1: build docker images from scratch
After cleaning up all images and everything from Docker locally I recreated the `db` image. 

As a result I obtain 39 tables:
![image](https://github.com/awslabs/aws-dataall/assets/71252798/c481ddb1-1f5a-405e-adfe-d90e94737eea)

If I now go to an actual deployment of data.all for the current `main` branch and check the amount of tables, I see that the aurora database contains 49 tables:

![image](https://github.com/awslabs/aws-dataall/assets/71252798/76ce545d-f412-4317-a18a-b18fb74206bf)
![image](https://github.com/awslabs/aws-dataall/assets/71252798/9f1b2577-d112-4a0c-ac85-2cb686bd4ef3)
![image](https://github.com/awslabs/aws-dataall/assets/71252798/99fcd13f-dcf6-48b0-a7d7-8a27fa7ce09b)

**Comment by dlpzx** [comment]
### Test 2: drop tables and re-create them using alembic locally
Using the commands provided in the `makefile` with a small modifications I am dropping the `dkrcompose` tables locally:
```
drop-tables: upgrade-pip install-backend
	pip install 'alembic'
	export envname=dkrcompose
	export PYTHONPATH=./backend && \
	python backend/migrations/drop_tables.py
```
Result:

**Comment by dlpzx** [comment]
I am picking up this issue again for with the current version of the code (2.2)

**Comment by dlpzx** [comment]
## 1. Start local dockerized application
- When we run `docker-compose up` the postgres container is created with no tables or schemas
- in the GraphQL container, `local_graphql_server.py`uses [sqlalchemy declarative_base](https://docs.sqlalchemy.org/en/13/orm/extensions/declarative/basic_use.html) to create all tables with this function: `Base.metadata.create_all(engine.engine)` --->**alembic scripts are not executed**

@zsaltys I do not need to run alembic when starting the docker containers. There might be mismatches in comparison with the alembic migration tables but I cannot see that any data.all functionalities are affected
Can you show me the tables that are missing?


## 2. Drop and rerun all migrations should work

## 3. More docs
https://awslabs.github.io/aws-dataall/deploy-locally/ should be updated to include explaining the alembic migrations and how to run them to update to the latest state and how to add new ones. --> fully agree. We can tackle this in a separate PR to the github pages
## 4. tests
It is already implemented in the db-migration tests in the quality gate: `deploy/stacks/pipeline.py:464`

**Comment by dlpzx** [comment]
## Updates from offline discussion
Action points for this issue:
- Design the best way to develop migration scripts locally
- Update documentation (README or GitHub pages) and add a piece on how to develop migration scripts locally
- Clean-up RDS tables that are no longer needed

**Comment by noah-paige** [comment]
Handled the above fix in #1033  - closing this issue, please re-open if any additional comments / concerns
